### PR TITLE
fix: Duration add/subtract

### PR DESCRIPTION
**Pull request by alexander-ulaev**
fix Duration add/subtract doesn't take into account how many days in a month #2336

### CODE REVIEW

File: src/plugin/duration/index.js [hunk 1]
```diff
@@ -257,6 +257,24 @@ class Duration {
...
 }
 
+const addDuration = (date, duration) =>
+  date.add(duration.years(), 'year')
+    .add(duration.months(), 'month')
+    .add(duration.days(), 'day')
+    .add(duration.hours(), 'hour')
+    .add(duration.minutes(), 'minute')
+    .add(duration.seconds(), 'second')
+    .add(duration.milliseconds(), 'millisecond')
```
Comment (iamkun) [inline src/plugin/duration/index.js:267]: a little trick to minimize the code, pass the third argument to combine add and subtract together.
```
// manipulateDuration(this, value, 1)
// manipulateDuration(this, value, -1)

// e.g. .add(duration.months() * (-1), 'month')
```

also, one thing that confused me, is why adding `value.asMilliseconds()`  is not the correct result. 
It should be the real `ms` between the two dates.

Comment (alexander-ulaev) [reply src/plugin/duration/index.js]: thanks for the tip to improve the code
value.asMilliseconds() executed without date context and In this case, month is equal to 30 days. I think it's correct. (moment js has same behavior)

File: src/plugin/duration/index.js [hunk 2]
```diff
@@ -257,6 +257,15 @@ class Duration {
...
 }
 
+const manipulateDuration = (date, duration, k) =>
+  date.add(duration.years() * k, 'year')
+    .add(duration.months() * k, 'month')
+    .add(duration.days() * k, 'day')
+    .add(duration.hours() * k, 'hour')
+    .add(duration.minutes() * k, 'minute')
+    .add(duration.seconds() * k, 'second')
+    .add(duration.milliseconds() * k, 'millisecond')
```
Comment (iamkun) [inline src/plugin/duration/index.js:267]: we can replace 'second' with 's' according to https://github.com/iamkun/dayjs/blob/dev/src/utils.js#L31

to save some bytes

**Comment by iamkun** [comment]
Cheers

**Pull request merged**